### DISCUSSION TITLE
Compatibility of Slavic languages with Western (Torus font)

### DISCUSSION THREAD

**Discussion by Loreos7** [ppy/osu#14679] [Q&A]
Is it possible to change font/weight of current slavic osu!lazer fonts for compatibility to torus font?
It looks kinda weird now while in osu!stable it looks almost the same
<img src="https://user-images.githubusercontent.com/86934170/132483674-79c3940a-f451-4a49-b3ff-4a6256649f50.png" width="350" />
<img src="https://user-images.githubusercontent.com/86934170/132483668-4200b2f7-8620-4eb6-a527-826efa6d4bbb.png" width="350" />

**Comment by frenzibyte** [comment]
This is actually a bug with `TextBuilder` not falling back to a font with the same specified weight, but falls back to the first font that owns a glyph for the requested of character in order of addition, and thus always falling back to the lowest weight of the font.

For reference, here's how it should've displayed:
![CleanShot 2021-09-08 at 20 32 39](https://user-images.githubusercontent.com/22781491/132557323-1ca02783-19b6-48e7-a7f9-f42ca1748aa4.png)

Though it still doesn't look quite well. Almost suggesting to force the `osu!` part to be displayed in the same font used for the other glyphs in that case, maybe.

**Reply by Loreos7** [reply]
<img src="https://user-images.githubusercontent.com/86934170/132577784-22c73374-57ec-42de-97ba-27ff7826bc7a.png" width="300" />
Looks better, but now smth with alignment
And there is difference btw weights (compare "o")

**Reply by bdach** [reply]
yes, that was what was meant by "doesn't look quite well".

(also please stop using markdown headings for large text. normal text is just as fine.)

**Reply by peppy** [reply]
@frenzibyte wanna open a tracking issue on o!f side if that doesn't exist already? seems like something we want to keep tabs on.

**Reply by frenzibyte** [reply]
yeah there doesn't seem to be one existing already, will do.

**Reply by frenzibyte** [reply]
https://github.com/ppy/osu-framework/issues/6595

**Comment by Loreos7** [comment]
@frenzibyte Hey, hmmm..
I know this is a low priority task but I just wanna say I found similar to `Torus Regular` free font which contains Russian glyphs so you maybe could apply it. Called `Comfortaa` designed by [Johan Aakerlund](https://fonts.google.com/?query=Johan%20Aakerlund)

![image](https://github.com/ppy/osu/assets/86934170/9615975a-b907-41d1-93c0-3b050e5fb87e)
![image](https://github.com/ppy/osu/assets/86934170/75f16d55-ee1d-47a2-ae76-ea360d49b551)

**Reply by Loreos7** [reply]
If you'll do it could actually close this issue for Russian, Belarus, Ukrainian, Serbian and Bulgarian languages.

**Reply by tadatomix** [reply]
Tbh, I'd offer different font. This one doen't give enough difference between latin and cyrillic characters. д, а, р, п, и, for example, should be much different from, g, a, p, n, u. If it could be possible to find another font, it would be better. Plus, this still doesn't close issues with Greek, Korean, Chinese (both Simplified and Traditional) and Japanese fonts. Hardest could be last 3, since not all people designing all of the existing kanjis (10k minimum, for japanese, +-50k for Chinese)

**Reply by tadatomix** [reply]
~~https://fonts.google.com/specimen/Zen+Maru+Gothic 
This might be used for Greek, Japanese and Cyrillic? I don't consider Chinese here, since it doesn't contain even some basic characters like 你 or 吗~~

Forget my words, it has different size, compared to what Torus has
![изображение](https://github.com/user-attachments/assets/2b4a1886-65ca-4083-9c95-6507529f01f4)

*Maybe the problem could be behind this https://github.com/ppy/osu/discussions/14679#discussioncomment-1297572*

**Reply by tadatomix** [reply]
TLDR: It's way too hard to find a good font. Maybe I'll try again, next time

**Decided to change all that, since I looked again, and Inter was used for any Cyrillic and Greek glyphs and Noto is for any Asian glyphs, so**

In the end, I think lazer needs a better fallback font than Inter and Noto, but the problem is that, so far, I can't find fonts, which could contain:

1. Font with Greek, Cyrillic and Cyrillic extended characters
2. Font with all of the Simplified and Traditional Chinese, plus Hiragana and Katakana, Thai and Korean characters 

_Funny enough, for some reason, half of Asian fonts contain Cyrillic glyphs, but don't contain all of the glyphs from both chinese languages :/_

Using different fonts for each case is quite worse, since it implies different designs and it would be obvious that way too much fonts were used for one line (e.g. in japanese, line could contain Hiragana, Katakana, Kanji and English at the same time, where Hiragana and Katakana grabs from font A, Kanji from font B and English from font C)
